src/inputs/TreeSelectField/TreeSelectField.tsx

```diff
@@ -392,71 +425,81 @@ function TreeSelectFieldBase<O, V extends Value>(props: TreeSelectFieldProps<O,
+            if (option && option.children && option.children.length > 0) {
+              const childrenKeys = option.children
+                .flatMap(flattenOptions)
+                .map((o) => valueToKey(getOptionValue(o)))
+                .filter((childKey) => {
+                  // remove children that are disabled
+                  return !state.disabledKeys.has(childKey);
+                });
+              [key, ...childrenKeys].forEach(addedKeys.add, addedKeys);
+            }
 
-          // If this was a child that was selected, then see if every other child is also selected, and if so, consider the parent selected.
-          // Walk up the parents and determine their state.
-          for (const parent of parents.reverse()) {
-            const allChecked = parent.children?.every((child) => {
-              const childKey = valueToKey(getOptionValue(child));
-              return addedKeys.has(childKey) || existingKeys.has(childKey);
-            });
-            if (allChecked) {
-              addedKeys.add(valueToKey(getOptionValue(parent)));
+            // If this was a child that was selected, then see if every other child is also selected, and if so, consider the parent selected.
+            // Walk up the parents and determine their state.
+            for (const parent of parents.reverse()) {
+              const allChecked = parent.children?.every((child) => {
+                const childKey = valueToKey(getOptionValue(child));
+                return addedKeys.has(childKey) || existingKeys.has(childKey) || state.disabledKeys.has(childKey);
+              });
+              if (allChecked) {
+                addedKeys.add(valueToKey(getOptionValue(parent)));
+              }
             }
           }
         });
 
         // For removed keys, we need to also unselect any children and parents of the removed key
         [...removedKeys].forEach((key) => {
           // Grab the option and parents of the option that was removed
-          const maybeOption = findOption(fieldState.allOptions, key, getOptionValue);
-          if (!maybeOption) return;
-
-          const { option, parents } = maybeOption;
-
-          // If the option has children, then we need to unselect those children as well
-          if (option.children && option.children.length > 0) {
-            const childrenKeys = option.children.flatMap(flattenOptions).map((o) => valueToKey(getOptionValue(o)));
-            [key, ...childrenKeys].forEach(removedKeys.add, removedKeys);
-          }
+          const maybeOptions = findOptions(fieldState.allOptions, key, getOptionValue);
+          if (maybeOptions.length === 0) return;
+
+          for (const { option, parents } of maybeOptions) {
+            // If the option has children, then we need to unselect those children as well
+            if (option.children && option.children.length > 0) {
+              // Ensure we do not impact children that are disabled. They should be able to toggled on/off
```

Comment:
question: "should *not* be able to be toggled"?

Reply:
hah yes! The comments are wrong but, I think the logic is right. I'm pretty sure I wrote a test for this. Will update the comment. Thanks!

Reply:
I don't have a test for this. I'll write one really quick.